### PR TITLE
Adding basic implementation of CreateTableFromSnapshot

### PR DESCRIPTION
**Pull request by roopak-qlogic**
Basic Implementation of CreateTableFromSnapshot, Unit test case creation is in progress

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-cpp/994)

### CODE REVIEW

**Comment by roopak-qlogic** [comment]
getting some issues during creation of unit test case, so currently working on that, mean time I pushed the basic API implementation code.

**Comment by roopak-qlogic** [comment]
It seems build has failed due to timeout and download issue, looking into it

**Comment by coryan** [comment]
The builds have been super flaky in the last few days.  I am restarting the ones that failed for you.

**Comment by roopak-qlogic** [comment]
One of the travis build failed with some timeout issue, may need to restart.

**Comment by paprat** [comment]
Hey roopak,

I restarted the build and it failed again because of this reason.

[WARNING] /v/third_party/googletest/googletest/src/gtest-death-test.cc:836:: Death tests use fork(), which is unsafe particularly in a threaded context. For this test, Google Test detected 2 threads.
Aborting because exceptions are disabled: unrecoverable error in MakeCall(): TableAdmin(parent=projects/the-project/instances/the-instance) uh-oh [7=PERMISSION_DENIED] -

Let me know if I should restart it again.

**Comment by roopak-qlogic** [comment]
PTAL

File: google/cloud/bigtable/internal/table_admin.h [hunk 1]
```diff
@@ -155,6 +156,55 @@ class TableAdmin {
 
   //@}
 
+  template <typename ResultType>
```
Comment (coryan) [inline google/cloud/bigtable/internal/table_admin.h]: I thought you were going to refactor this code, will that happen in a future PR?  If so, please add a TODO with a bug number to track that work.

File: google/cloud/bigtable/internal/table_admin.h [hunk 2]
```diff
@@ -155,6 +156,55 @@ class TableAdmin {
...
+      std::this_thread::sleep_for(delay);
+      google::longrunning::GetOperationRequest op;
+      op.set_name(operation.name());
+      grpc::ClientContext context;
+      status = client_->GetOperation(&context, op, &operation);
+      if (not status.ok() and not polling_policy->OnFailure(status)) {
+        return ResultType{};
+      }
+    } while (not polling_policy->Exhausted());
+    std::ostringstream os;
```
Comment (coryan) [inline google/cloud/bigtable/internal/table_admin.h]: This code is not tested, or at least codecov.io thinks it is not, consider adding a test that exhaust the polling policy.

File: google/cloud/bigtable/internal/table_admin.h [hunk 3]
```diff
@@ -155,6 +156,55 @@ class TableAdmin {
...
+               << ") - "
+               << "invalid result type in operation=" << operation.name();
+            status = grpc::Status(grpc::StatusCode::UNKNOWN, os.str());
+            return ResultType{};
+          }
+          ResultType result;
+          any.UnpackTo(&result);
+          return result;
+        }
+        if (operation.has_error()) {
```
Comment (coryan) [inline google/cloud/bigtable/internal/table_admin.h]: This branch is never tested:

https://codecov.io/gh/GoogleCloudPlatform/google-cloud-cpp/pull/994/diff#D9-180

Consider a test that intentionally returns an error.

**Comment by roopak-qlogic** [comment]
One of the build is failing, but not getting, is it related to my change or  need to restart the build.

**Comment by coryan** [comment]
@roopak-qlogic I restarted that build.  Please let me know when this PR is ready for another review.  I still think the tests do not cover enough of the `PollLongRunningOperation()` function

**Comment by roopak-qlogic** [comment]
Did not get chance to add new test cases, but in next couple of days I will add and push for review.

**Comment by roopak-qlogic** [comment]
Code is ready for review

**Comment by roopak-qlogic** [comment]
PTAL

**Pull request merged**